src/shims/panic.rs

```diff
@@ -161,7 +161,24 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
     }
 
-    /// Starta a panic in the interpreter with the given message as payload.
+    /// Handler that should be called when unsupported functionality is encountered.
+    /// This function will either panic within the context of the emulated application
+    /// or return an error in the Miri process context
+    ///
+    /// Return value of `Ok(bool)` indicates whether execution should continue.
+    fn handle_unsupported<S: AsRef<str>>(&mut self, error_msg: S) -> InterpResult<'tcx, ()> {
+        let error = format!("unsupported Miri functionality: {}", error_msg.as_ref());
+
+        if self.eval_context_ref().machine.panic_on_unsupported {
+            // message is slightly different here to make automated analysis easier
+            self.start_panic(error.as_ref(), StackPopUnwind::Skip)?;
+            return Ok(());
+        } else {
+            throw_unsup_format!("{}", error);
```

Comment:
Here it should literally use `error_msg`, not the `error` with the prefix. Error rendering adds its own prefix.